harvard-lts/mps-embed
# Add passing tests. Add GitHub Action that runs unit tests.

**Add passing tests. Add GitHub Action that runs unit tests.**
* * *

**JIRA Tickets**: [LTSVIEWER-103](https://jira.huit.harvard.edu/browse/LTSVIEWER-103) [LTSVIEWER-100](https://jira.huit.harvard.edu/browse/LTSVIEWER-100)

* Other Relevant Links:
  - [MPS Asset Delivery Images Testing Scripts](https://github.huit.harvard.edu/LTS/mps-asset-delivery-images/blob/main/app/package.json#L9-L12)
  - [MPS Asset Delivery Images Test Folder](https://github.huit.harvard.edu/LTS/mps-asset-delivery-images/tree/main/app/__tests__)
  - [SysDev Automated Tests Wiki](https://wiki.harvard.edu/confluence/pages/viewpage.action?spaceKey=LibraryTechServices&title=SysDev+-+Automated+tests)

# What does this Pull Request do?
This PR adds Unit Tests and Integration Tests for Embed. I based this off of the examples provided in the MPS Asset Delivery Images repository and the SysDev Automated Tests Wiki (both linked above in the Relevant Links). There is also a new GitHub Action that just runs the Unit tests. The Github action passed when I deployed to this branch in [Job #1](https://github.com/harvard-lts/mps-embed/runs/4873682541?check_suite_focus=true). 

# How should this be tested?

A description of what steps someone could take to:
* Build and run the `mps-embed` container off of the `add-real-tests` branch
* ssh into the mps-embed container: `docker exec -it mps-embed bash`
* Run each of the test scripts and confirm they all pass. The available test scripts are:
  - `npm run test` (This runs both unit and integration tests)
  - `npm run test:unit` (This runs only the unit tests)
  - `npm run test:integration` (This runs only the integration tests)
  - `npm run test:coverage` (This runs all the tests and creates a `/coverage/` folder with coverage reports)
  - `npm run test:badges`(This runs all the tests and creates badge image files in the `/coverage/` folder)

**NOTE:** The `/coverage` folder is in `.gitignore` so the files are not committed to the Github repository. You can look at the report it generates by opening the local index file in your browser (the application does not have a "route" to the report.  The index file is located at: `/mps-viewer/coverage/lcov-report/index.html`. You can see the badge images it creates a similar way with `/mps-viewer/coverage/badge-branches.svg` and the other .svg files located in the `/coverage` folder. Integration of the badges and the repo's `README.md` coming soon in a different PR.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? Yes
- integration tests? Yes

# Interested parties
@awoods @enriquediaz
